File: 12_ChocolatesByNumbers.cpp
# include <cmath>

int solution(int N, int M)
{
	long int i{1};

	while (1)
	{
		if (N>M)
			if (fmod(i*N,M)==0)
				return i*N/M;
			else
				i++;
		else
			if (fmod(i*M,N)==0)
				return i;
			else
				i++;
	}
	return 0;
}

File: 12_ChocolatesByNumbers.py
def solution(N, M):
	
	i=1

	while (True):
		if (N>M):
			x=i*N//M
			if i*N-x*M ==0:
				return x
			else:
				i+=1
		else:
			x=i*M//N
			if i*M-x*N ==0:
				return i
			else:
				i+=1
	return 0


